lib/Dialect/HW/Transforms/LowerHWtoBTOR2.cpp

```diff
@@ -0,0 +1,786 @@
+  const static constexpr StringLiteral sdiv = "sdiv";
+  const static constexpr StringLiteral udiv = "udiv";
+  const static constexpr StringLiteral smod = "smod";
+  const static constexpr StringLiteral concat = "concat";
+  const static constexpr StringLiteral notStr = "not";
+  const static constexpr StringLiteral neq = "neq";
+  const static constexpr StringLiteral hwNeq = "ne";
+  const static constexpr StringLiteral ite = "ite";
+  const static constexpr StringLiteral implies =
+      "implies";                                        // logical implication
+  const static constexpr StringLiteral state = "state"; // Register state
+  const static constexpr StringLiteral next =
+      "next"; // Register state transition
+  const static constexpr StringLiteral bad = "bad";
+  const static constexpr StringLiteral constraint = "constraint";
+  const static constexpr StringLiteral ws = " ";  // WhiteSpace
+  const static constexpr StringLiteral nl = "\n"; // NewLine
+
+  /// Field helper functions
+
+  // Checks if a constant of a given size has been declared
+  // If so, its lid will be returned
+  // Otherwise -1 will be returned
+  size_t getConstLID(int64_t val, size_t w) {
+    // Look for the pair
+    for (auto p : constToLIDMap) {
+      // Compare the given value and width to those in the entry
+      if ((val == p.getFirst().first) && (w == p.getFirst().second)) {
+        return p.getSecond();
+      }
+    }
+    // if no lid was found return -1
+    return NO_LID;
+  }
+
+  // Updates or creates an entryfor  a constant of a given size
+  // associating it with the current lid
+  void setConstLID(int64_t val, size_t w) {
+    // Create a new entry for the pair
+    constToLIDMap[std::make_pair<int64_t, size_t>(
+        std::forward<int64_t>(val), std::forward<size_t>(w))] = lid;
+  }
+
+  // Checks if a sort was declared with the given width
+  // If so, its lid will be returned
+  // Otherwise -1 will be returned
+  size_t getSortLID(size_t w) {
+    // Look for presence of the width in a pair
+    for (auto p : sortToLIDMap) {
+      // If the width was declared, return it's lid
+      if (w == p.first) {
+        return p.second;
+      }
+    }
+    // If no lid was found return -1
+    return NO_LID;
+  }
+
+  // Updates or creates an entry for the given width
+  // associating it with the current lid
+  void setSortLID(size_t w) {
+    // Check for the existence of the sort
+    for (size_t i = 0; i < sortToLIDMap.size(); ++i) {
+      auto p = sortToLIDMap[i];
+
+      // If the width was declared, update it's lid
+      if (w == p.first) {
+        sortToLIDMap[i] = std::make_pair<size_t, size_t>(
+            std::forward<size_t>(w), std::forward<size_t>(lid));
+        return;
+      }
+    }
+
+    // Otherwise simply create a new entry
+    sortToLIDMap.push_back(std::make_pair<size_t, size_t>(
+        std::forward<size_t>(w), std::forward<size_t>(lid)));
```

Comment:
I'm not sure what you think std::forward is suppose to do here, but you don't need it and I suspect every place you use it is unnecessary.